### PR TITLE
Add current domain condition

### LINKED ISSUE #190: Add current domain condition plugin

### PR DESCRIPTION
**Pull request by artreaktor**
Fix #190

### CODE REVIEW

File: domain/src/Plugin/Condition/CurrentDomain.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+use Drupal\Core\Plugin\Context\ContextDefinition;
+
+/**
+ * Provides a 'Current domain' condition.
+ *
+ * @Condition(
+ *   id = "current_domain",
+ *   label = @Translation("Current domain"),
+ *   context = {
+ *     "domain" = @ContextDefinition("entity:domain", label = @Translation("Domain"))
```
Comment (agentrickard) [inline domain/src/Plugin/Condition/CurrentDomain.php]: Malformed annotation

File: domain/src/Plugin/Condition/CurrentDomain.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $this->configuration['current_domain'] = array_filter($form_state->getValue('current_domain'));
+    parent::submitConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
```
Comment (agentrickard) [inline domain/src/Plugin/Condition/CurrentDomain.php:67]: How does this interact with domain.js?

**Comment by agentrickard** [comment]
I don't understand the use-case for this code. Where is it visible in the UI? Why is the existing ContextProvider not sufficient?

**Comment by artreaktor** [comment]
What's wrong with annotation? I used Drupal\domain\Plugin\Condition\Domain.php as an example.

**Comment by agentrickard** [comment]
I see. That annotation is actually a method signature, so one line is ok.

See Drupal\node\Plugin\Condition\NodeType

**Comment by agentrickard** [comment]
OK, so we don't need a next Condition handler, we need to make sure that the existing one \Drupal\domain\Plugin\Condition\Domain functions properly for Blocks and Panels, yes?

**Comment by artreaktor** [comment]
+1

**Pull request closed without merging**